sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+## API Report File for "@azure/synapse-spark"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { HttpResponse } from '@azure/core-http';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+// @public (undocumented)
+export type CancelSparkBatchJobOptions = OperationOptions;
+
+// @public (undocumented)
+export type CancelSparkSessionOptions = OperationOptions;
+
+// @public (undocumented)
+export type CancelSparkStatementOptions = OperationOptions;
+
+// @public
+export interface CreateSparkBatchJobOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type CreateSparkBatchJobResponse = SparkBatchJob & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJob;
+    };
+};
+
+// @public
+export interface CreateSparkSessionOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type CreateSparkSessionResponse = SparkSession & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type CreateSparkStatementOptions = OperationOptions;
+
+// @public
+export type CreateSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public
+export interface GetSparkBatchJobOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkBatchJobResponse = SparkBatchJob & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJob;
+    };
+};
+
+// @public
+export interface GetSparkSessionOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type GetSparkSessionResponse = SparkSession & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type GetSparkStatementOptions = OperationOptions;
+
+// @public
+export type GetSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public (undocumented)
+export interface ListPageSettings {
+    continuationToken?: string;
+}
+
+// @public
+export interface ListSparkBatchJobsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
```

Comment:
when making this a pageable, you should encode this parameter (and possibly size) into an opaque continuation token, e.g. https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/search/search-documents/src/searchClient.ts#L634

Reply:
I agree and understand your point, but that parameter is required for our service, these are also follow the disign of Livy, user can get specified size and offsert of jobs.

---
In reply to: [521608118](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521608118) [](ancestors = 521608118)

Reply:
So you are saying that a continuation token doesn't make sense because you expect users to manually manipulate pagination parameters?

Reply:
I think so, let's keep it as so at beta.1 first. 

---
In reply to: [524823795](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r524823795) [](ancestors = 524823795)